Modix.Services/IsUp/IsUpService.cs

```diff
@@ -0,0 +1,35 @@
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Discord.Commands;
+using Newtonsoft.Json;
+
+namespace Modix.Services.IsUp
+{
+    public class IsUpService
+    {
+        private const string _apiBaseURl = "https://isitdown.site/api/v3/";
+
+        public IHttpClientFactory HttpClientFactory { get; }
+
+        public IsUpService(IHttpClientFactory httpClientFactory)
+        {
+            HttpClientFactory = httpClientFactory;
+        }
+
+        public async Task<IsUpResponse> GetIsUpResponseAsync([Summary("Url to get status of")]string url)
+        {
+            var apiQueryUrl = $"{_apiBaseURl}{url}";
+            var client = HttpClientFactory.CreateClient();
+            var response = await client.GetAsync(apiQueryUrl);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                return null;
```

Comment:
Instead of returning null, would it make more sense to return an object? For example, if it's a BadRequest, usually there's a reason in the response. (though it doesn't look like their api is that sophisticated)

Reply:
I can change it to an error object, there does seem to be a failure reason/statuscode